feat: filter shops and order status in dashboard

futurepaul opened the pull request:
closes #685 

two caveats:
I can't find a good way to make the dropdown the same width as the button. w-full overflows the screen. I can maybe look deeper into how dropdowns work to see if I can come up with a fix but I think it's separate from this pr anyway
There's some layout jitter when it switches to no orders, I believe that's the ordertable's fault for overflowing. I see in the figma there's a different look for order list items so perhaps that can be the solve?

https://github.com/user-attachments/assets/178405eb-1e1c-4be5-ab1c-73e0fc933d2b

plebomoto commented:
Nice work! 

Have you considered using the [select](https://www.shadcn-svelte.com/docs/components/select) component? It seems to have the proper width out of the box.

Also lets have the filter in shops and products too.

futurepaul commented:
> Have you considered using the [select](https://www.shadcn-svelte.com/docs/components/select) component? It seems to have the proper width out of the box.

Nice, this works way better. I updated the styling a bit to closer match the figma (@Zazawowow mentioned not liking the checkboxes for instance)

> Also lets have the filter in shops and products too.

I added the shops filter to products, not sure what to filter for the shops page

<img width="561" alt="Screenshot 2025-02-20 at 9 59 58 AM" src="https://github.com/user-attachments/assets/7c333c99-1332-428f-9896-1c5ff3e06c6d" />
<img width="573" alt="Screenshot 2025-02-20 at 9 59 49 AM" src="https://github.com/user-attachments/assets/af03c9a5-4187-439a-ab0e-a23909f700bc" />
<img width="696" alt="Screenshot 2025-02-20 at 9 59 44 AM" src="https://github.com/user-attachments/assets/762744ef-4ad0-49b6-ad34-0fa921838040" />

plebomoto commented:
Looks great. Product filtering works exactly as expected.

Could you take a look at why the sales don't respond?


https://github.com/user-attachments/assets/1047fe21-992c-4ffa-b479-463d2f9efd1c

gzuuus commented:
Looking great! Eventually we can use https://tanstack.com/virtual/latest/docs/introduction which is an amazing lib for virtual list with sorting functionallities built in. I was using it in another project and its pretty good

futurepaul commented:
> Could you take a look at why the sales don't respond?

great catch @plebomoto, this was working fine with a smaller data set but the bigger data broke the reactivity for order table. should be fixed now

https://github.com/user-attachments/assets/f1f6471d-17d3-4f6e-8c87-3d40d6511cfb
